src/clincoded/upgrade/resourceHistory.py

```diff
@@ -0,0 +1,41 @@
+from contentbase.upgrader import upgrade_step
+
+@upgrade_step('snapshot', '1', '2')
+def snapshot_1_2(value, system):
+    # https://github.com/ClinGen/clincoded/issues/2132
+    # Remove existing provisional/snapshot nested data
+
+    try:
+        if (value['resourceType'] == 'classification'):
+            curation_object_name = 'gdm'
+            provisional_array_name = 'provisionalClassifications'
+            snapshot_array_name = 'associatedClassificationSnapshots'
+        elif (value['resourceType'] == 'interpretation'):
+            curation_object_name = 'interpretation'
+            provisional_array_name = 'provisional_variant'
+            snapshot_array_name = 'associatedInterpretationSnapshots'
+        else:
+            raise KeyError
+
+        del value['resource'][snapshot_array_name]
+
+        try:
+            for provisional in value['resourceParent'][curation_object_name][provisional_array_name]:
+                try:
+                    for snapshot in provisional[snapshot_array_name]:
+                        try:
+                            del snapshot['resource'][snapshot_array_name]
+
+                        except KeyError:
+                            pass
+
+                except KeyError:
+                    pass
+
+        except KeyError:
+            pass
+
+    except KeyError:
+        pass
```

Comment:
I would suggest using just one single `try...except KeyError` (the most outer try-except, in this case), if we're just `pass` for all `KeyError`s, since the most outer try-except will catch any first `KeyError`, if any occurs. Not necessarily a big deal so I'm going to approve this PR anyway.

Reply:
The `try` clause on line 22 (and corresponding `except` clause on line 35) is probably unnecessary, but the others (inside the `for` loops) are needed in order to process all array elements (vs. breaking out of the loop on the first `KeyError`).

Reply:
Oh I see! That makes a lot of sense.